core/sail/nativerdf/src/main/java/org/eclipse/rdf4j/sail/nativerdf/btree/BTree.java

```diff
@@ -292,6 +292,12 @@ public BTree(File dataDir, String filenamePrefix, int blockSize, int valueSize,
 			this.valueSize = buf.getInt();
 			this.rootNodeID = buf.getInt();
 
+			if (rootNodeID == 0) {
+				if (nioFile.size() > blockSize) {
```

Comment:
This is a somewhat brittle approach and it's really only relevant when using the soft fail property. So maybe this should be wrapped with the property too and throw something other than an illegal state exception.